src/index/am.rs

```diff
@@ -292,27 +297,90 @@ pub unsafe extern "C" fn amrescan(
     _n_orderbys: std::os::raw::c_int,
 ) {
     unsafe {
-        if (*scan).numberOfKeys != 0 {
-            pgrx::error!("vector search with attributes is not supported");
-        }
-        if (*scan).numberOfOrderBys == 0 {
-            pgrx::error!("vector search without a ORDER BY clause is not supported");
+        if (*scan).numberOfOrderBys < 0 || (*scan).numberOfOrderBys > 1 {
+            pgrx::error!(
+                "vector search with {} ORDER BY clauses is not supported",
+                (*scan).numberOfOrderBys
+            );
         }
-        if (*scan).numberOfOrderBys != 1 {
-            pgrx::error!("vector search with too many ORDER BY clauses is not supported");
+        if (*scan).numberOfKeys < 0 {
```

Comment:
1. No need to check it.
2. You must check `== 0 && (*scan).numberOfOrderBys == 0`.